apps/choleskey/cholesky_tiled.cpp

```diff
@@ -0,0 +1,161 @@
+                    const int lapackLay) {
+    unsigned int m = 0, n = 0, k = 0;
+
+    for (k = 0; k < num_tiles; ++k) {
+        //POTRF
+        int info = LAPACKE_dpotrf(LAPACK_ROW_MAJOR, 'L', tile_size, matrix_split[k * num_tiles + k], tile_size);
+
+        for (m = k + 1; m < num_tiles; ++m) {
+            //DTRSM
+            cblas_dtrsm(blasLay, CblasRight, CblasLower, CblasTrans, CblasNonUnit, tile_size, tile_size, 1.0,
+                        matrix_split[k * num_tiles + k], tile_size, matrix_split[m * num_tiles + k], tile_size);
+        }
+
+        for (n = k + 1; n < num_tiles; ++n) {
+            //DSYRK
+            cblas_dsyrk(blasLay, CblasLower, CblasNoTrans, tile_size, tile_size, -1.0, matrix_split[n * num_tiles + k],
+                        tile_size, 1.0, matrix_split[n * num_tiles + n], tile_size);
+
+            for (m = n + 1; m < num_tiles; ++m) {
+                //DGEMM
+                cblas_dgemm(blasLay, CblasNoTrans, CblasTrans, tile_size, tile_size, tile_size, -1.0,
+                            matrix_split[m * num_tiles + k], tile_size, matrix_split[n * num_tiles + k], tile_size, 1.0,
+                            matrix_split[m * num_tiles + n], tile_size);
+            }
+        }
+    }
+}
+
+int main(int argc, char** argv) {
+
+    // parse params
+    args_params_t args = argparse::parse<args_params_t>(argc, argv);
+    int tile_size = 0, tot_tiles = 0;
+
+    std::uint64_t matrix_size = args.mat_size;  // Number of matrix dimension.
+    std::uint64_t num_tiles = args.num_tiles;   // matrix size MUST be divisible
+    int verifycorrectness = args.verifycorrectness;
+
+    bool layRow = true;
+
+    fmt::print("matrix_size = {}, num_tiles = {}\n\n", matrix_size, num_tiles);
+
+    // Check mat_size is divisible by num_tiles
+    if (matrix_size % num_tiles != 0) {
+        fmt::print("matrix size must be divisible by num_tiles.. aborting\n");
+        throw std::invalid_argument("Matrix size is not divisible by num_tiles");
+    }
+
+    if (matrix_size == 0) {
+        fmt::print(" 0 illegal input matrix_size \n");
+        std::exit(0);
+    }
+
+    tile_size = matrix_size / num_tiles;
+    tot_tiles = num_tiles * num_tiles;
+
+    data_type* A = new data_type[matrix_size * matrix_size];
+
+    // Allocate memory for tiled_cholesky for the full matrix
+    data_type* A_lower = new data_type[matrix_size * matrix_size];
+
+    // Allocate memory for MKL cholesky for the full matrix
+    data_type* A_MKL = new data_type[matrix_size * matrix_size];
+
+    // Memory allocation for tiled matrix
+    data_type** Asplit = new data_type*[tot_tiles];
+
+    for (int i = 0; i < tot_tiles; ++i) {
+        // Buffer per tile
+        Asplit[i] = new data_type[tile_size * tile_size];
+    }
+
+    //Generate a symmetric positve-definite matrix
+    A = generate_positiveDefinitionMatrix(matrix_size);
+
+    CBLAS_ORDER blasLay;
```

Comment:
put a comment here explain what is CBLAS_ORDER type is.

Reply:
added in code.